src/Dompdf.php

```diff
@@ -1363,6 +1355,10 @@ public function getDom()
      */
     public function setOptions(Options $options)
     {
+        if ($this->options && $this->options->getHttpContext() && !$options->getHttpContext()) {
+            $options->setHttpContext($this->options->getHttpContext());
+        }
+
```

Comment:
I think I would leave this out, as none of the other options are merged like this.

Reply:
I think the same, buy i trying to avoid breaking changes, if you do this
https://github.com/dompdf/dompdf/blob/e76193db8a7f14de95f78888757e0c167929af3a/tests/DompdfTest.php#L38-L39
`$dompdf->setOptions(new Options());`  overwrites `setHttpContext`, i try to has the same functionality as before

Reply:
Right, hadn’t thought of that. Then I would add a short comment explaining why the logic is there.

Reply:
Maybe simply `// For backwards compatibility` would be enough. Just so it is easier to decide whether to remove the logic in a future major release (or deliberately keep it).

Reply:
added